### DISCUSSION TITLE
没有生成ingressclasses资源

### DISCUSSION THREAD

**Discussion by borederman** [higress-group/higress#451] [Q&A]
我如何通过yaml生成 higress的 ingressclass，`spec.controller` 应该填什么？

通过 helm 安装之后，是没有创建 ingressclass资源的，虽然并不影响在ingress资源里，指定ingressclass为 higress。

```
# kubectl get ingressclasses.networking.k8s.io
NAME    CONTROLLER             PARAMETERS   AGE
nginx   k8s.io/ingress-nginx   <none>       461d
```

我对比了nignx-ingress-controller和higress-contorller的区别。
nignx-ingress-controller 启动参数有一项 `--controller-class=k8s.io/ingress-nginx`，而higress-contorller并没有类似配置。

**Comment by johnlanni** [comment]
是的 这块可以扩展支持下，你有兴趣实现下吗，可以记录个issue

**Comment by johnlanni** [comment]
之前觉得实现这个有点多余，直接通过name关联就可以了，但发现有些平台会强行校验这个ingressclass资源是否存在

**Comment by CH3CHO** [comment]
https://github.com/alibaba/higress/issues/468

**Comment by srepancake** [comment]
Error from server (invalid ingress class: IngressClass.networking.k8s.io "higress" not found): error when creating "aws-ingress.yaml": admission webhook "vingress.elbv2.k8s.aws" denied the request: invalid ingress class: IngressClass.networking.k8s.io "higress" not found

在AWS EKS上还是需要这个资源的，无法直接做到从ingress-nginx到higress的无缝迁移

**Reply by johnlanni** [reply]
@Libertad-904483782 可以先手动创建下，后面会支持装一个自带的
